replacement_scan.go

```diff
@@ -0,0 +1,65 @@
+package duckdb
+
+/*
+   #include <stdlib.h>
+   #include <duckdb.h>
+   void replacement_scan_cb(duckdb_replacement_scan_info info, const char *table_name, void *data);
+   typedef const char cchar_t;
+*/
+import "C"
+import (
+	"sync"
+	"unsafe"
+)
+
+type ReplacementScanCallback func(tableName string) (string, []any, error)
+
+var replacementScanFnMxt sync.Mutex
+var replacementScanCallback ReplacementScanCallback
+
+func RegisterReplacementScan(connector *Connector, cb ReplacementScanCallback) error {
+	replacementScanFnMxt.Lock()
+	replacementScanCallback = cb
+	replacementScanFnMxt.Unlock()
+	C.duckdb_add_replacement_scan(connector.db, C.duckdb_replacement_callback_t(C.replacement_scan_cb), nil, C.duckdb_delete_callback_t(C.free))
+	return nil
+}
+
+//export replacement_scan_cb
+func replacement_scan_cb(info C.duckdb_replacement_scan_info, table_name *C.cchar_t, data *C.void) {
+	replacementScanFnMxt.Lock()
+	scanner := replacementScanCallback
+	replacementScanFnMxt.Unlock()
```

Comment:
To clarify, this synchronises replacing the currently registered replacement scan with a new one?

Reply:
That was my thought. It gives the caller more control over how to deal with errors, on the other hand it as annoying if concurrent libraries introduces it's own replacement. 
I think it is better to let the client decide how to deal with concurrent use of replacement.

Reply:
Yes, I agree. Moving this to the application level makes more sense to me... So could we take it out of this PR? I would still keep the `var` wrapper, even if the struct then only contains a single element.